// нет, не эквивалентны. В первом примере catch позволит обработать ошибку из f1, во втором ошибку из f1 ничто уже не обработает
